Socket.lean

```diff
@@ -329,19 +249,16 @@ instance : Coe SockAddrUnix SockAddr where
   coe sa := .unixAddr sa
 
 alloy c extern "lean_sockaddr_in_family"
-def SockAddr4.family (addr : @& SockAddr4) : AddressFamily := {
-  return (lean_to_sockaddr_in(lean_ctor_get(addr, 0)))->sin_family;
-}
+def SockAddr4.family (addr : @& SockAddr4) : AddressFamily :=
+  return of_lean<SockAddr4>(lean_ctor_get(addr, 0))->sin_family;
```

Comment:
Also, something I noticed will writing this code: Is the `lean_ctor_get` meant to be here? The `SocketAddr*` specializations do not have fields, so I would think these would just be `of_lean<SockAddr*>(addr)->s*_family`.

Reply:
Oh you are right on this one, that's a copy paste mistake back from when I still had this function on `SockAddr`, can you fix it as well please?